internal/restconfig/restconfig.go

```diff
@@ -213,6 +205,19 @@ func (rcp *Producer) RunOnSelectedContext(function PerContextFn, status reporter
 		return status.Error(err, "error building the cluster.Info for the default configuration")
 	}
 
+	var submVersion string
+	if clusterInfo.Submariner != nil {
+		submVersion = clusterInfo.Submariner.Spec.Version
+	} else if clusterInfo.ServiceDiscovery != nil {
+		submVersion = clusterInfo.ServiceDiscovery.Spec.Version
+	}
+
+	if submVersion != "" {
+		if err = checkVersionMismatch(submVersion); err != nil {
+			return status.Error(err, "versions mismatched")
```

Comment:
Minor thing but `checkVersionMismatch` already returns a suitable user-friendly message - there's no need to augment it here.

```suggestion
			return status.Error(err, "")
```